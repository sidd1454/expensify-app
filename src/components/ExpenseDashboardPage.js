import React from 'react';
import ExpenseList from './ExpenseList';
import ExpenseListFilters from './ExpenseListFilters';

class ExpenseDashboardPage extends React.Component {
    constructor(props){
        super(props);
        this.ExpenseDashboardPage = this.ExpenseDashboardPage.bind(this);
    }
}

ExpenseDashboardPage = () => (
    <div>
        <ExpenseListFilters />
        <ExpenseList />
    </div>
);

export default ExpenseDashboardPage;