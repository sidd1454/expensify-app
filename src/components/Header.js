import React from 'react';
import { NavLink } from 'react-router-dom';

class Header extends React.Component {
    constructor(props){
        super(props);
        this.Header = this.Header.bind(this);
    }
}

Header = () => (
    <header>
        <h1>Expensify</h1>
        <NavLink to="/" activeClassName="is-active" exact={true}>Dashboard</NavLink>
        <NavLink to="/create" activeClassName="is-active"> Create Expense</NavLink>
    </header>
);

export default Header;