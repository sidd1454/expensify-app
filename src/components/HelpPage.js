import React from 'react';

class HelpPage extends React.Component {
    constructor(props){
        super(props);
        this.HelpPage = this.HelpPage.bind(this);
    }
}

HelpPage = () => (
    <div>
        This is Help page
    </div>
);

export default HelpPage;