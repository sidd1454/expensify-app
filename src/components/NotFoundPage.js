import React from 'react';
import { Link } from 'react-router-dom';

class NotFoundPage extends React.Component {
    constructor(props){
        super(props);
        this.NotFoundPage = this.NotFoundPage.bind(this);
    }
}

NotFoundPage = () => (
    <div>
        404 - <Link to="/">Go Home</Link>
    </div>
);

export default NotFoundPage;