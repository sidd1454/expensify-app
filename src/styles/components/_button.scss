.button {
    background: $blue;
    color: white;
    border: none;
    display: inline-block;
    text-decoration: none;
    padding: $s-size;
    font-weight: 300;
    line-height: 1;
    font-size: $font-size;
}

.button--link {
    background: none;
}

.button--secondary {
    @extend .button;
    background-color: #888888;
}