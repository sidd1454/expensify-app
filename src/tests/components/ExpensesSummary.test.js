import React from 'react';
import { shallow } from 'enzyme';
import { ExpensesSummary } from '../../components/ExpensesSummary';

test('should correctly render expenses summary with expense ', () => {
    const wrapper = shallow(<ExpensesSummary expenseCount={1} expensesTotal={125}/>);
    expect(wrapper).toMatchSnapshot();
});

test('should correctly render multiple expenses summary with expense ', () => {
    const wrapper = shallow(<ExpensesSummary expenseCount={23} expensesTotal={125511548}/>);
    expect(wrapper).toMatchSnapshot();
});