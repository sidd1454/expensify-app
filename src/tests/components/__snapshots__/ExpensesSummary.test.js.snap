// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`should correctly render expenses summary with expense  1`] = `
ShallowWrapper {
  "length": 1,
  Symbol(enzyme.__root__): [Circular],
  Symbol(enzyme.__unrendered__): <ExpensesSummary
    expenseCount={1}
    expensesTotal={125}
/>,
  Symbol(enzyme.__renderer__): Object {
    "batchedUpdates": [Function],
    "getNode": [Function],
    "render": [Function],
    "simulateEvent": [Function],
    "unmount": [Function],
  },
  Symbol(enzyme.__node__): Object {
    "instance": null,
    "key": undefined,
    "nodeType": "host",
    "props": Object {
      "children": <h1>
        Viewing 
        1
         
        expense
         totalling 
        $1.25
</h1>,
    },
    "ref": null,
    "rendered": Object {
      "instance": null,
      "key": undefined,
      "nodeType": "host",
      "props": Object {
        "children": Array [
          "Viewing ",
          1,
          " ",
          "expense",
          " totalling ",
          "$1.25",
        ],
      },
      "ref": null,
      "rendered": Array [
        "Viewing ",
        1,
        " ",
        "expense",
        " totalling ",
        "$1.25",
      ],
      "type": "h1",
    },
    "type": "div",
  },
  Symbol(enzyme.__nodes__): Array [
    Object {
      "instance": null,
      "key": undefined,
      "nodeType": "host",
      "props": Object {
        "children": <h1>
          Viewing 
          1
           
          expense
           totalling 
          $1.25
</h1>,
      },
      "ref": null,
      "rendered": Object {
        "instance": null,
        "key": undefined,
        "nodeType": "host",
        "props": Object {
          "children": Array [
            "Viewing ",
            1,
            " ",
            "expense",
            " totalling ",
            "$1.25",
          ],
        },
        "ref": null,
        "rendered": Array [
          "Viewing ",
          1,
          " ",
          "expense",
          " totalling ",
          "$1.25",
        ],
        "type": "h1",
      },
      "type": "div",
    },
  ],
  Symbol(enzyme.__options__): Object {
    "adapter": ReactSixteenAdapter {
      "options": Object {
        "enableComponentDidUpdateOnSetState": true,
      },
    },
  },
}
`;

exports[`should correctly render multiple expenses summary with expense  1`] = `
ShallowWrapper {
  "length": 1,
  Symbol(enzyme.__root__): [Circular],
  Symbol(enzyme.__unrendered__): <ExpensesSummary
    expenseCount={23}
    expensesTotal={125511548}
/>,
  Symbol(enzyme.__renderer__): Object {
    "batchedUpdates": [Function],
    "getNode": [Function],
    "render": [Function],
    "simulateEvent": [Function],
    "unmount": [Function],
  },
  Symbol(enzyme.__node__): Object {
    "instance": null,
    "key": undefined,
    "nodeType": "host",
    "props": Object {
      "children": <h1>
        Viewing 
        23
         
        expenses
         totalling 
        $1,255,115.48
</h1>,
    },
    "ref": null,
    "rendered": Object {
      "instance": null,
      "key": undefined,
      "nodeType": "host",
      "props": Object {
        "children": Array [
          "Viewing ",
          23,
          " ",
          "expenses",
          " totalling ",
          "$1,255,115.48",
        ],
      },
      "ref": null,
      "rendered": Array [
        "Viewing ",
        23,
        " ",
        "expenses",
        " totalling ",
        "$1,255,115.48",
      ],
      "type": "h1",
    },
    "type": "div",
  },
  Symbol(enzyme.__nodes__): Array [
    Object {
      "instance": null,
      "key": undefined,
      "nodeType": "host",
      "props": Object {
        "children": <h1>
          Viewing 
          23
           
          expenses
           totalling 
          $1,255,115.48
</h1>,
      },
      "ref": null,
      "rendered": Object {
        "instance": null,
        "key": undefined,
        "nodeType": "host",
        "props": Object {
          "children": Array [
            "Viewing ",
            23,
            " ",
            "expenses",
            " totalling ",
            "$1,255,115.48",
          ],
        },
        "ref": null,
        "rendered": Array [
          "Viewing ",
          23,
          " ",
          "expenses",
          " totalling ",
          "$1,255,115.48",
        ],
        "type": "h1",
      },
      "type": "div",
    },
  ],
  Symbol(enzyme.__options__): Object {
    "adapter": ReactSixteenAdapter {
      "options": Object {
        "enableComponentDidUpdateOnSetState": true,
      },
    },
  },
}
`;
